src/openapi/transport/auth.ts

```diff
@@ -0,0 +1,224 @@
+            result.url,
+            oldTokenExpiry,
+        );
+
+        if (areUrlAuthErrorsProblematic) {
+            // Blocking infinite loop of authorization re-requests which might be caused by invalid
+            // behavior of given endpoint which constantly returns 401 error.
+            log.error(
+                LOG_AREA,
+                'Too many authorization errors occurred for different tokens within a specified time-frame for a specific endpoint',
+                result.url,
+            );
+            throw {
+                message: 'Auth overload',
+                isNetworkError: false,
+            };
+        }
+
+        log.debug(LOG_AREA, 'Authentication failure', result);
+
+        this.authProvider.tokenRejected(oldTokenExpiry);
+    }
+    throw result;
+}
+
+// -- Exported methods section --
+
+/**
+ * This class builds on top of {@link saxo.openapi.TransportCore} and adds authentication management. You need only
+ * construct one or the other, they automatically wrap each other. All of the options from the {@link saxo.openapi.TransportCore}
+ * constructor are valid here as well.
+ * For authentication management, this class will wait until just before the authentication expires (see tokenRefreshMarginMs)
+ * and will refresh the token generating an event which is picked up by some of the other Transports.
+ * @class
+ * @alias saxo.openapi.TransportAuth
+ * @param {string} baseUrl - The base url used for all open api requests. This should be an absolute URL.
+ * @param {Object} authProvider - a AuthProvider to get the token from
+ * @param {Object} [options] - Options for auth and for the core transport. See Transport.
+ * @param {string} [options.language] - The language sent as a header if not overridden.
+ * @param {boolean} [options.defaultCache=true] - Sets the default caching behaviour if not overridden on a call.
+ * @param {number} [options.authErrorsDebouncePeriod] - The period within which errors on different tokens cause an endpoint auth errors
+ *                                                      to be ignored.
+ */
+class TransportAuth {
+    authErrorsDebouncePeriod = DEFAULT_AUTH_ERRORS_DEBOUNCE_PERIOD;
+    authorizationErrors: Record<
+        string,
+        Array<{ authExpiry: number; added: number }>
+    > = {};
+
+    // its a timeout id
+    errorCleanupTimeOutId: any;
+    // needs to map with transport core interface
+    transport: any;
+    authProvider: AuthProvider;
+    constructor(
+        baseUrl: string,
+        authProvider: AuthProvider,
+        options?: Options,
+    ) {
+        if (!authProvider) {
+            throw new Error('transport auth created without a auth provider');
+        }
+
+        if (options?.authErrorsDebouncePeriod) {
+            this.authErrorsDebouncePeriod = options.authErrorsDebouncePeriod;
+        }
+
+        // @ts-ignore fix-me
+        this.transport = new TransportCore(baseUrl, options);

```

Comment:
Seems like we need to extend the Option type to include all possible options, looks like the jsdoc is missing some properties

Reply:
@hiredgun , should we mark as fix me for now, until we get to know all possible options with integration to saxotrader

Reply:
Can't we figure them out from the options required in TransportCore? IMHO it would be better to do it now otherwise we may forget it.